File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> m;
        for(auto &e:nums){
            if(m[e] == 1){
                return true;
            }
            m[e]++;
        }
        return false;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.py
import heapq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = []

        for e in nums:
            if len(heap) == k:
                if heap[0] < e:
                    heapq.heappop(heap)
                    heapq.heappush(heap, e)
            else:
                heapq.heappush(heap, e)
        
        return heap[0]
